lib/facter/custom_facts/util/normalization.rb

```diff
@@ -16,16 +16,14 @@ class NormalizationError < StandardError; end
       # @return [void]
       def normalize(value)
         case value
-        when Integer, Float, TrueClass, FalseClass, NilClass, Symbol, Date
```

Comment:
I'm not sure we should be this permissive. I think now facter 4 cli will behave differently than it does in facter 3.

Reply:
Yep, I think the same.
I tried to make it simmilar to how facter 3 returns objects in irb:
```
2.7.2 :008 > Facter.add(:dd) do
2.7.2 :009 >   setcode { StandardError.new("test") }
2.7.2 :010 > end
 => #<Facter::Util::Fact:0x00007fe07f137ec0>
2.7.2 :011 > Facter.value(:dd)
 => #<StandardError: test>
2.7.2 :012 > Facter.value(:dd).class
 => StandardError
2.7.2 :013 >
```
 Also created https://github.com/puppetlabs/facter/pull/2314 just to add Date and Time in case anyone relied on only these types.